sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/DecryptOptions.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.keys.cryptography;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * A class containing various configuration parameters that can be applied when performing decryption operations.
+ */
+public class DecryptOptions {
+    /**
+     * Initialization vector to be used in the decryption operation using a symmetric algorithm.
+     */
+    @JsonProperty(value = "iv")
+    private final byte[] iv;
+
+    /**
+     * Get additional data to authenticate when performing decryption with an authenticated algorithm.
+     */
+    @JsonProperty(value = "aad")
+    private final byte[] additionalAuthenticatedData;
+
+    /**
+     * The tag to authenticate when performing decryption with an authenticated algorithm.
+     */
+    @JsonProperty(value = "tag")
+    private final byte[] authenticationTag;
+
+    /**
+     * Creates an instance of {@link DecryptOptions} with the given parameters.
+     *
+     * @param iv Initialization vector for symmetric algorithms.
+     * @param additionalAuthenticatedData Additional data to authenticate but not encrypt/decrypt when using
+     * authenticated crypto algorithms.
+     * @param authenticationTag The tag to authenticate when performing decryption with an authenticated algorithm.
+     */
+    public DecryptOptions(byte[] iv, byte[] additionalAuthenticatedData, byte[] authenticationTag) {
```

Comment:
This requires them knowing what to pass for which algorithms. This is not a good customer UX. Please see https://apiview.dev/Assemblies/Review/d74052731cb94759804beaa492b7a17a#Azure.Security.KeyVault.Keys.Cryptography.DecryptOptions. Perhaps a builder would be more idiomatic, but the overall concept is to make sure they can only provide relevant arguments that also dictates the algorithm, rather than pairing them together correctly. We can make crypto easier in our limited APIs (compared to general crypto provided in our respective frameworks).

Reply:
Will evaluate a different approach. Probably what you suggested [here](https://github.com/Azure/azure-sdk-for-java/pull/17209#discussion_r521590535).